engine/stagemaster/src/lib.rs

```diff
@@ -156,118 +169,131 @@ pub fn setup(
     system: &mut ActorSystem,
     renderables: Vec<RenderableID>,
     env: &'static environment::Environment,
-    window: &GlutinFacade,
+    window: WindowBuilder,
 ) {
     ::monet::setup(system);
 
-    system.add(UserInterface::new(window.clone()), move |mut the_ui| {
+    let context = ContextBuilder::new().with_vsync(true);
+    let events_loop = EventsLoop::new();
+    let display = Display::new(window, context, &events_loop).unwrap();
+
+
+    system.add(UserInterface::new(display.clone(), events_loop), move |mut the_ui| {
         let ui_id = the_ui.world().id::<UserInterface>();
 
         let mut scene = SceneDescription::new(renderables.clone().into());
         scene.eye.position *= 30.0;
         let renderer_id = RendererID::spawn(
-            External::new(window.clone()),
+            External::new(display.clone()),
             vec![scene].into(),
             &mut the_ui.world(),
         );
 
-        use monet::{ProjectionRequesterID, MSG_ProjectionRequester_projected_3d};
+        use monet::MSG_ProjectionRequester_projected_3d;
 
         the_ui.on_critical(move |_: &ProcessEvents, ui, world| {
-            let scale = ui.imgui.display_framebuffer_scale();
+            let scale = ui.ui_inner.imgui.display_framebuffer_scale();
 
-            for event in ui.window.poll_events().collect::<Vec<_>>() {
+            let ui_inner = &mut ui.ui_inner;
+
+            ui.events_loop.poll_events(|event| {
                 match event {
```

Comment:
Again, can we match `event.event` to keep the match arms simpler?

Reply:
fixed, thanks!